test/integ/test_arrow_result.py

```diff
@@ -198,6 +206,15 @@
 ]
 
 
+@contextmanager
+def structured_type_wrapped_conn(conn_cnx):
+    with conn_cnx() as conn:
+        if STRUCTURED_TYPES_SUPPORTED:
+            for param in STRUCTURED_TYPE_PARAMETERS:
+                conn.cursor().execute(f"alter session set {param}=true").fetchall()
+        yield conn
```

Comment:
nit: do you need to unset it afterwards?

Reply:
I believe the session is destroyed once the `with conn_cnx() as conn` scope is left so it should not matter.

Reply:
Out of curiosity - do you test with native arrow only?

Reply:
if so, there is an easier way you can do it like this: https://github.com/snowflakedb/snowflake-connector-python/blob/3013fe884a0bf794663ab6e3c18299d90d50095e/test/integ/test_cursor.py#L1566-L1569

Reply:
There are other tests that verify that json results work. In my experiments it looked like a structured response returned in json format and a semi structured response returned in json format were the same so I haven't added tests here. I don't think it would be hard to expand these tests to those cases though.

Reply:
I'm not sure if I agree. We had a discussion with Berlin team about nulls in arrays - currently nulls in arrays are returned as `undefined` instead of `null`. It breaks JSON parsing in drivers, so there is a small chance that for structured types some time in the future it may be returned differently comparing to Arrow.

But the question is still valid - your parameters enabled native arrow, so in that case this is not a JSON anymore. We have three formats to return structured types:
- Response format = JSON (JSON data where each structured type is a separate string containing JSON)
- Response format = Arrow (we have a string column and each row contains a separate string containing JSON - so technically this is similar to JSON)
- Response format = Arrow AND native Arrow enabled (full arrow - if we have a structured object, it is represented as arrow struct column with a pack of arrow columns representing each column)

Reply:
That sounds worth having explicit tests for. I'll look in to those cases and add relevant tests as part of another task.